### PR TITLE
Add section for static file server

### PR DESCRIPTION
**Pull request by iceiilin**
Add section for static file server configuration

http://10.62.59.222:8000/rackhd/static_file_server.html is the document

@cgx027 @lanchongyizu

### CODE REVIEW

File: docs/rackhd/static_file_server.rst [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+.. _on-http: https://github.com/RackHD/on-http
+.. _nginx: https://www.nginx.com/
+.. _install nginx: https://www.nginx.com/resources/wiki/start/topics/tutorials/install/
+.. _nginx_conf: https://www.nginx.com/resources/wiki/start/topics/examples/full/
+.. _config.json: https://github.com/RackHD/RackHD/blob/master/packer/ansible/roles/monorail/files/config.json
+.. _httpProxies: http://rackhd.readthedocs.io/en/latest/rackhd/configuration.html?highlight=httpProxies
+.. _httpStaticRoot: http://rackhd.readthedocs.io/en/latest/rackhd/configuration.html?highlight=httpStaticRoot
+
+
+There are two kinds of static files in RackHD, and this section introduces a mechanism to move one type of
```
Comment (cgx027) [inline docs/rackhd/static_file_server.rst]: Might be good to give a very short introduction to those two types of static files, or pointing user to a link that introduces those two types.

File: docs/rackhd/static_file_server.rst [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+same static file directory).  `on-http`_ manages all the files mentioned above by default, and the latter
+(**files for discovery and OS installation**) can be moved to a third-party static file server, which will be
+discussed below.
+
+Diagrams for Different Working Modes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+RackHD supports three modes to serve static files.
+
+Legacy Mode: nodes get static files from `on-http`_ service (default).
+Single-Host Mode: nodes get static files from another service in the same host as RackHD.
```
Comment (cgx027) [inline docs/rackhd/static_file_server.rst]: Although the reStruct file looks like a new line for 'Single-Host Mode' but the rendered html shows it's in the same line as 'Legacy Mode'. Suggest to give a new line, maybe even add bullets :)

File: docs/rackhd/static_file_server.rst [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    "fileServerPath": '/',
+    ...
+
+"fileServerAddress" is the IP address of static file server that nodes can access.
+"fileServerPort" is the port the server is listening to (*optional*, the default value is 80).
+"fileServerPath" is the "location" in server configuration (*optional*, the default value is '/').
+
+Restart RackHD services after adding these fields.
+
+**Note**: fileServer configuration takes higher priority than `httpProxies`_ and 'httpStaticRoot'_. which means that when above fields exists,
```
Comment (cgx027) [inline docs/rackhd/static_file_server.rst]: Configuration -> configurations.

**Comment by lanchongyizu** [comment]
:+1:

**Comment by anhou** [comment]
:+1:

**Pull request merged**